version: '3.7'
services:
  ultron8_ci:
    container_name: ultron8_ci
    privileged: true
    build:
      context: .
      dockerfile: Dockerfile
      # target: runtime-image
      # FIXME: https://github.com/docker/compose/issues/5090
      # FIXME: https://github.com/docker/docker-py/issues/998
      cache_from:
        - bossjones/ultron8-ci:runtime-image
      args:
        CONTAINER_GID: ${CONTAINER_GID:-1000}
        CONTAINER_UID: ${CONTAINER_UID:-1000}
    image: bossjones/ultron8-ci:runtime-image
    ports:
      - 11267:11267
    environment:
      - CONTAINER_GID=${CONTAINER_GID:-1000}
      - CONTAINER_UID=${CONTAINER_UID:-1000}
      - TRAVIS
      - TRAVIS_BRANCH
      - TRAVIS_BUILD_DIR
      - TRAVIS_JOB_ID
      - TRAVIS_JOB_NUMBER
      - TRAVIS_PULL_REQUEST
      - TRAVIS_COMMIT
      - TRAVIS_REPO_SLUG
      - TRAVIS_OS_NAME
      - TRAVIS_TAG
      - ULTRON_ENABLE_WEB=${ULTRON_ENABLE_WEB:-false}
      - BETTER_EXCEPTIONS=1
    tty: true
    stdin_open: true
    entrypoint:
      - bash
    command: /home/developer/app/.ci/pytest_runner.sh
    user: ${CONTAINER_UID:-1000}:${CONTAINER_GID:-1000}
    volumes:
      - .:/home/developer/app:rw
      - ~/.cache/pip:/home/developer/.cache/pip
      - ~/.wheelhouse/:/home/developer/.wheelhouse/
    # ports:
    #   - ${PORT}:3000
